python/ray/remote_function.py

```diff
@@ -103,7 +103,9 @@ def __init__(
         # When gpu is used, set the task non-recyclable by default.
         # https://github.com/ray-project/ray/issues/29624 for more context.
         num_gpus = self._default_options.get("num_gpus") or 0
-        if num_gpus > 0 and self._default_options.get("max_calls", None) is None:
+        if (
+            num_gpus > 0 and self._default_options.get("max_calls", None) is None
+        ) or "nsight" in (self._default_options.get("runtime_env") or {}):
```

Comment:
Maybe we don't need this change?

Reply:
I added this in order to ensure worker process is not reused for Ray task since by default, nsight profiler generates report when the process exit.

Reply:
I see. Can you add a comment here? We should also add this to the doc when we write one